thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/api/user/dashboard/UserDashboardResource.java

```diff
@@ -242,6 +258,35 @@ public UserDashboardResource(MergedAnomalyResultManager anomalyDAO, MetricConfig
     return output;
   }
 
+  private Collection<MergedAnomalyResultDTO> fetchAnomaliesByMetricDatasetPairs(Long start, Long end, String metricsRef) {
+    List<MergedAnomalyResultDTO> anomalies = new ArrayList<>();
+
+    if (StringUtils.isBlank(metricsRef)) {
+      return Collections.emptyList();
+    }
+
+    List<Predicate> predicates = new ArrayList<>();
+    predicates.add(Predicate.GE("endTime", start));
+    if (end != null) {
+      predicates.add(Predicate.LT("startTime", end));
+    }
+
+    String[] metricDatasetPairs = metricsRef.split(",");
+    for (String metricDatasetPair : metricDatasetPairs) {
+      List<Predicate> metricDatasetPred = new ArrayList<>(predicates);
```

Comment:
can put the parsing logic into a factory method of the static inner class and JAX-RS will handle this automatically. For example
`
private static class MetricDatasetPair { 

    String metricName;

    String datasetName;

    public static MetricDatasetPair fromString(String input){

     // parse the input strting and initilize the object

    }
}
`